hbase-server/src/main/java/org/apache/hadoop/hbase/wal/RecoveredEditsOutputSink.java

```diff
@@ -89,20 +89,68 @@ private RecoveredEditsWriter getRecoveredEditsWriter(TableName tableName, byte[]
 
   @Override
   public List<Path> close() throws IOException {
-    boolean isSuccessful = true;
+    boolean isSuccessful;
     try {
       isSuccessful = finishWriterThreads();
-    } finally {
-      isSuccessful &= closeWriters();
+    } catch (InterruptedIOException e) {
+      // TODO if it's caught, should we log the error? If it's not rethrown?
+      abortWriters();
+      // TODO should the exception be rethrown again here to bubble up?
```

Comment:
I would suggest re-throw, since that is what the previous code was implicitly doing.